src/v/cluster/security_frontend.cc

```diff
@@ -255,18 +273,38 @@ ss::future<std::vector<delete_acls_result>> security_frontend::do_delete_acls(
      */
     auto removed_bindings = _authorizer.local().remove_bindings(filters, true);
 
-    delete_acls_cmd_data data;
-    data.filters = std::move(filters);
-    delete_acls_cmd cmd(std::move(data), 0 /* unused */);
+    auto should_sanction
```

Comment:
A delete request is a vector of filters, any of which could (as I understand it) match _both_ role ACLs and user ACLs, so I was thinking about a case like that where you might want to reject only the role-oriented part of some match.

But it's too specific as a justification anyway - more generally I think any change to this model requires rethinking the whole API surface, which feels beyond the scope of this PR.

>in the future...make erroring more specific for both the create and delete ACLs endpoints

I do think we should.

Reply:
> might want to reject only the role-oriented part of some match

I think we should treat each delete filter as a unit and each delete acl filter should pass/fail independently. We could reject a delete filter if it matched any Role-bound ACLs. I think we would be able to do that relatively easily because `removed_bindings` is a vector of vectors where at index `i` we have a vector of ACL bindings that matched the delete filter `i`. And the response is a vector of errors, so we could error the delete filters independently. Am I missing something here?

Reply:
Yeah that all sounds technically possible. My concern here is that, currently, security_frontend does this:

https://github.com/redpanda-data/redpanda/blob/79757b61af9e1f187b6cc511c7388c1e074bcb70/src/v/cluster/security_frontend.cc#L258-L283

which means that a single error code (returned by the controller) is propagated to each filter's corresponding `delete_acls_result`. As I understand it, this is due to some limitation in the interface between controller frontend & backend.

Given the time constraint, I think it's in our interest to just sort of reproduce those semantics here. Namely that a request to security_frontend either completely succeeds (all changes applied to controller log) or completely fails (no changes applied to controller log). Relaxing this guarantee should be possible but imposes an additional downstream validation cost.

TL;DR - Given time constraint, exercise caution

Reply:
FYI I got some clarity here, we will permit ACL deletion